http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeoAuthProvider.scala

```diff
@@ -65,6 +65,16 @@ abstract class LeoAuthProvider[F[_]] {
                                    googleProject: GoogleProject,
                                    runtimeName: RuntimeName)(implicit ev: ApplicativeAsk[F, TraceId]): F[Boolean]
 
+  /**
+   *
+   *
+   * @param internalId
+   * @return returns a list of Cluster Actions that the user has access to
+   */
+  def getNotebookClusterActions(internalId: RuntimeInternalId, userInfo: UserInfo)(
+    implicit ev: ApplicativeAsk[F, TraceId]
+  ): F[List[String]]
```

Comment:
Perhaps this should return a `List[NotebookClusterAction]` instead of a `List[String]`